traefikee/templates/proxy/deployment.yaml

```
@@ -1,72 +1,76 @@
+{{- $proxies := dict "proxy" .Values.proxy }}
+{{- $proxies = merge $proxies .Values.additionalProxies }}
+{{- range $proxyName, $proxy := $proxies }}
 ---
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ .Values.cluster  }}-proxy
-  namespace: {{ .Release.Namespace }}
+  name: {{ $.Values.cluster  }}-{{ $proxyName }}
+  namespace: {{ $.Release.Namespace }}
+  keys: {{ $proxies }}
   labels:
-    {{- include "common.labels" . | nindent 4 }}
-    {{- with .Values.proxy.deploymentLabels }}
+    {{- include "common.labels" $ | nindent 4 }}
+    {{- with $proxy.deploymentLabels }}
     {{- toYaml . | nindent 4 }}
     {{- end }}
   annotations:
-    {{- with .Values.proxy.deploymentAnnotations }}
+    {{- with $proxy.deploymentAnnotations }}
     {{- toYaml . | nindent 4 }}
     {{- end }}
 spec:
-  {{- with .Values.proxy.replicas }}
+  {{- with $proxy.replicas }}
   replicas: {{ . }}
   {{- end }}
   selector:
     matchLabels:
       app: traefikee
-      release: {{ .Values.cluster }}
+      release: {{ $.Values.cluster }}
       component: proxies
```

I think the selector should be more accurate, don't you think ? 
```suggestion
      component: proxies
      proxy: {{ $proxyName }}
```